### PR TITLE
feat(biome_graphql_parser): parse union type definition

### PR DESCRIPTION
**Pull request by vohoanglong0107**
## Summary

Parse Graphql Union Type definition ([link](https://spec.graphql.org/October2021/#UnionTypeDefinition)). This is a small case so I included both the happy path parsing and the error handling.

## Test Plan

All tests should pass

### CODE REVIEW

File: crates/biome_graphql_parser/src/parser/definitions/union.rs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        p.bump(T![|]);
+    }
+
+    let position = p.source().position();
+    UnionMemberTypeList.parse_list(p);
+
+    // has not progressed, meaning no union member types were parsed
+    if position == p.source().position() {
+        p.error(expected_named_type(p, p.cur_range()));
+    }
```
Comment (vohoanglong0107) [inline crates/biome_graphql_parser/src/parser/definitions/union.rs:65]: Is there a better way to check if `UnionMemberTypeList` was empty?

Comment (ematipico) [reply crates/biome_graphql_parser/src/parser/definitions/union.rs:65]: Internally, ParsedLists, check the progression of the parser and they fail if that parser hasn't advanced.

Check the source code to make sure

**Comment by vohoanglong0107** [comment]
Yeah, this is exactly the approach used by parse_list. However, parse_list
uses ParseProgress, which has a private constructor and panic if the parser
doesn't progress, so I just simplified it to this.

File: crates/biome_graphql_parser/src/parser/definitions/union.rs (hunk 1 above)
Comment (vohoanglong0107) [reply crates/biome_graphql_parser/src/parser/definitions/union.rs:65]: Well, this is exactly the approach used by parse_list. However, parse_list uses ParseProgress, which has a private constructor and panic if the parser doesn't progress, so I just simplified it to this.

**Review by ematipico — APPROVED** [review]

**Pull request merged**